### PR TITLE
JTAG: Use new withClock way of overriding clocks

### PR DESCRIPTION
**Pull request by mwachs5**
the override clock way is deprecated

### CODE REVIEW

**Comment by ducky64** [comment]
Does this compile? The NegativeEdgeLatch constructor has changed to not take block, but your `new NegativeEdgeLatch` is still being passed a clock and data type. And without the need to be at a module boundary to change clock domains, I'm not convinced a separate NegativeEdgeLatch class is needed, the NegativeEdgeLatch object can just create a new clock scope and instantiate the registers inside.

Also, this probably should be an upstream change in the chisel-jtag repo (along with some other general cleanup work there).

**Comment by mwachs5** [comment]
Sorry no it doesn't compile, i got called into a meeting right after making this change, will fix it up.

**Comment by mwachs5** [comment]
(I will squash all these)

File: src/main/scala/jtag/Utils.scala [hunk 1]
```diff
@@ -12,40 +13,23 @@ class Tristate extends Bundle {
...
-  val reg = Reg(dataType)
-  when (io.enable) {
-    reg := io.next
-  }
-  io.output := reg
-}
-
 /** Generates a register that updates on the falling edge of the input clock signal.
   */
 object NegativeEdgeLatch {
```
Comment (aswaterman) [inline src/main/scala/jtag/Utils.scala]: This isn't germane to your pull request, but shouldn't this be called NegativeEdgeReg (or perhaps NegEdgeReg)?  It's not level-sensitive, so Latch confused me.

File: src/main/scala/jtag/Utils.scala [hunk 2]
```diff
@@ -12,40 +13,23 @@ class Tristate extends Bundle {
...
+    }
   }
 }
 
 /** A module that counts transitions on the input clock line, used as a basic sanity check and
   * debug indicator clock-crossing designs.
   */
-class ClockedCounter(modClock: Clock, counts: BigInt, init: Option[BigInt])
-    extends Module(override_clock=Some(modClock)) {
+class ClockedCounter(counts: BigInt, init: Option[BigInt]) extends Module {
```
Comment (aswaterman) [inline src/main/scala/jtag/Utils.scala]: Now that we're using withClock, is this any different than using Counter(true.B, counts) from Chisel?

Comment (aswaterman) [reply src/main/scala/jtag/Utils.scala]: Oh, I guess the ability to set the reset value to something nonzero is a difference.

**Comment by mwachs5** [comment]
I am holding off on merging this until it is actually tested, presumably with #1088

**Pull request merged**